Rubberduck.Core/UI/CodeExplorer/Commands/RemoveCommand.cs

```diff
@@ -31,33 +43,57 @@ protected override void OnExecute(object parameter)
                 return;
             }
 
-            var qualifiedModuleName = node.Declaration.QualifiedName.QualifiedModuleName;
-            var message = string.Format(CodeExplorerUI.ExportBeforeRemove_Prompt, node.Name);
+            RemoveComponent(node.Declaration.QualifiedModuleName);
+        }
 
-            switch (MessageBox.Confirm(message, CodeExplorerUI.ExportBeforeRemove_Caption, ConfirmationOutcome.Yes))
+        public bool RemoveComponent(QualifiedModuleName qualifiedModuleName, bool promptToExport = true)
+        {
+            if (promptToExport && !TryExport(qualifiedModuleName))
             {
-                case ConfirmationOutcome.Yes:
-                    if (!PromptFileNameAndExport(qualifiedModuleName))
-                    {
-                        return;
-                    }
-                    break;
-                case ConfirmationOutcome.No:
-                    break;
-                case ConfirmationOutcome.Cancel:
-                    return;
-                }
-
+                return false;
+            }
+            
             // No file export or file successfully exported--now remove it
-            var components = ProjectsProvider.ComponentsCollection(qualifiedModuleName.ProjectId);
             try
             {
-                components?.Remove(ProjectsProvider.Component(qualifiedModuleName));
+                _projectsRepository.RemoveComponent(qualifiedModuleName);
             }
             catch (Exception ex)
             {
-                MessageBox.NotifyWarn(ex.Message, string.Format(CodeExplorerUI.RemoveError_Caption, qualifiedModuleName.ComponentName));
+                _messageBox.NotifyWarn(ex.Message, string.Format(CodeExplorerUI.RemoveError_Caption, qualifiedModuleName.ComponentName));
+                return false;
             }
+
+            return true;
+        }
+
+        private bool TryExport(QualifiedModuleName qualifiedModuleName)
+        {
+            var projectId = qualifiedModuleName.ProjectId;
+            var projectType = _projectsRepository.Project(projectId).Type;
+            var component = _projectsRepository.Component(qualifiedModuleName);
+
+            if (projectType == ProjectType.HostProject && component.IsSaved)
+            {
+                return true; // File already up-to-date
+            }
+
+            var message = string.Format(CodeExplorerUI.ExportBeforeRemove_Prompt, qualifiedModuleName.Name);
+
+            switch (_messageBox.Confirm(message, CodeExplorerUI.ExportBeforeRemove_Caption, ConfirmationOutcome.Yes))
+            {
+                case ConfirmationOutcome.No:
+                    return true;
```

Comment:
this could use a bit of an explanation. e.g. `// user didn't want to import, nothing to do`

Reply:
Done